### PR TITLE
Gacha+Asset+User Settings+Card Text

### LINKED ISSUE #178: Add Song Filter to Official Art Assets + Sources In Search
Currently, you can link official art to a song, but you can't filter by it!
image

It's fairly simple to add, so I don't see why we shouldn't include it.

Also, I think that Sources should be searchable in the search bar. You COULD include source link too, but the source name usually will include stuff from the actual link, so that isn't a biggie. Some people might want to see how many assets we got from the pico website, or similar. it'd give them another option that should hopefully lead them to the assets they're searching for!

### LINKED ISSUE #179: Add Login/Title Screen Official Art To User Background Settings
Currently, users can select a background from the Background type of assets. But we are storing some things in Backgrounds that are actually just title screens that are also stored in official art.

Let's stop having duplicate images where they are not needed, and take advantage of the asset tagging system. Since the only official art tagged "Login" are title screens, let's add that to the selection of backgrounds for users too! We can then remove/replace the title screens in Background, keeping it a more accurate display overall!

### LINKED ISSUE #180: Allow Card Title/Skill Name Showing None W/O English Title
image
^ This occurs when the jp text is entered, but not en. Since we almost always get new info from JP, this'll pop up a lot unless we insert some conditionals in the code.

Should be easy to fix luckily, but the sooner done the better.

### PR DESCRIPTION
**Pull request by LatidoReMe**
+ Appends Gacha to the end of Gacha titles! Standardizes the view + checks if ガチャ is already at the end of JP gachas in case we forget to remove it or added it previously
+ Links Favorite Band to either 🇦 Related Members `/members/?i_band={}` OR 🇧 Songs from the Band `/songs/?i_band={}`
+ Adds is_song filter for official art and adds Source + Source Link to search, with adjusted search_field_labels so it'll not end in ... ( close #178 )
+ ⚙ Generates background settings to include Official Art tagged "Login"! Lets users continue to use title screens as bg's w/o storing them in the wrong places ( close #179 )
+ Adjusts card title and skill name to not show **None** just because we're missing one of the two translations! (close #180 )

### CODE REVIEW

File: bang/magicollections.py [hunk 1]
```diff
@@ -1610,11 +1609,13 @@ def to_fields(self, view, item, in_list=False, exclude_fields=None, *args, **kwa
             'korean_image': staticImageURL('language/kr.png'),
         }, exclude_fields=exclude_fields, **kwargs)
         if get_language() == 'ja' or unicode(item.t_name) == unicode(item.japanese_name):
-            setSubField(fields, 'name', key='value', value=item.japanese_name)
+            setSubField(fields, 'name', key='value', value=_('{} Gacha').format(
+                item.japanese_name[:-3] if item.japanese_name.endswith(u'ガチャ') else item.japanese_name))
```
Comment (summertriangle-dev) [inline bang/magicollections.py]: This check should be done when the gacha is added to the database, i.e. in the GachaForm.

You can do that pretty easily by adding a `def clean_japanese_name(self):` to GachaForm that removes `ガチャ` if present. We will need to check every existing gacha, but having consistent data is better in the long run.

File: bang/magicollections.py [hunk 2]
```diff
@@ -710,29 +712,26 @@ def to_fields(self, item, extra_fields=None, exclude_fields=None, order=None, *a
...
-            value = item.japanese_name
-            extra_fields.append(('card_name', {
-                'verbose_name': _('Title'),
-                'icon': 'id',
-                'type': 'title_text' if unicode(title) != unicode(value) else 'text',
-                'title': title,
-                'value': value,
-            }))
+            #Add Title
+            title=item.names.get(language, item.name if language not in settings.LANGUAGES_CANT_SPEAK_ENGLISH else None)
```
Comment (summertriangle-dev) [inline bang/magicollections.py]: Code style: spaces around `=`.

File: bang/magicollections.py [hunk 3]
```diff
@@ -710,29 +712,26 @@ def to_fields(self, item, extra_fields=None, exclude_fields=None, order=None, *a
...
+            value = item.japanese_name if item.japanese_name != None else item.name
+            if value != None:
```
Comment (summertriangle-dev) [inline bang/magicollections.py]: Also code style: test for None using `is`/`is not`. (Also for the line above this.)

File: bang/magicollections.py [hunk 4]
```diff
@@ -710,29 +712,26 @@ def to_fields(self, item, extra_fields=None, exclude_fields=None, order=None, *a
...
-            # Add skill name
-            if item.t_skill_name or item.japanese_skill_name:
-                title = item.skill_names.get(
-                    language, item.japanese_skill_name
-                    if language in settings.LANGUAGES_CANT_SPEAK_ENGLISH else item.skill_name)
-                value = item.japanese_skill_name
+            #Add Skill Name
+            title=item.skill_names.get(language, item.skill_name if language not in settings.LANGUAGES_CANT_SPEAK_ENGLISH else None)
+            value = item.japanese_skill_name if item.japanese_skill_name != None else item.skill_name
+            if value != None:
```
Comment (summertriangle-dev) [inline bang/magicollections.py]: Ditto for these three lines.

File: bang/magicollections.py [hunk 5]
```diff
@@ -710,29 +712,26 @@ def to_fields(self, item, extra_fields=None, exclude_fields=None, order=None, *a
...
-            extra_fields.append(('card_name', {
-                'verbose_name': _('Title'),
-                'icon': 'id',
-                'type': 'title_text' if unicode(title) != unicode(value) else 'text',
-                'title': title,
-                'value': value,
-            }))
+            #Add Title
+            title = item.names.get(language, item.name if language not in settings.LANGUAGES_CANT_SPEAK_ENGLISH else None)
+            value = item.japanese_name if item.japanese_name is not None else item.name
```
Comment (summertriangle-dev) [inline bang/magicollections.py]: i forgot to mention, this type of check can be simplified down to

```python
item.japanese_name or item.name
```

which takes the left hand side if it's nonempty, otherwise right.

File: bang/magicollections.py [hunk 6]
```diff
@@ -710,29 +712,26 @@ def to_fields(self, item, extra_fields=None, exclude_fields=None, order=None, *a
...
 
-            # Add skill name
-            if item.t_skill_name or item.japanese_skill_name:
-                title = item.skill_names.get(
-                    language, item.japanese_skill_name
-                    if language in settings.LANGUAGES_CANT_SPEAK_ENGLISH else item.skill_name)
-                value = item.japanese_skill_name
+            #Add Skill Name
+            title = item.skill_names.get(language, item.skill_name if language not in settings.LANGUAGES_CANT_SPEAK_ENGLISH else None)
+            value = item.japanese_skill_name if item.japanese_skill_name is not None else item.skill_name
```
Comment (summertriangle-dev) [inline bang/magicollections.py]: same here

File: bang/forms.py [hunk 7]
```diff
@@ -600,6 +600,9 @@ def __init__(self, *args, **kwargs):
         if 'c_versions' in self.fields:
             del(self.fields['c_versions'])
 
+    def clean_japanese_name(self):
+        return self.cleaned_data.get('japanese_name')[:-3] if self.cleaned_data.get('japanese_name').endswith(u'ガチャ') else self.cleaned_data.get('japanese_name')
```
Comment (summertriangle-dev) [inline bang/forms.py]: it would be cleaner to save the japanese_name in a variable so you don't have to .get() it every time.
